### PR TITLE
:sparkles: set user agent for vimclient

### LINKED ISSUE #1805: CAPV should set user agent on vimclient to aid transparency and quick troubleshooting
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

In current implementation, the user agent is not set for vimclient used to communicate with vCenter. It defaults to "Go-http-client/1.1" which most other components such as vsphere CPI, template-reslover-controller does not set as well.
When user list sessions using `govc session.ls`,  all it can see is lots of sessions initiated from control plane to vc, like:
```bash
kubo@wupYYmIJv24PJ:~$ govc session.ls|grep Go
5212f486-a546-4e2f-ffd7-ce4b044fbaa4  VSPHERE.LOCAL\Administrator                                           2023-02-27 22:52  old     10.168.194.127  Go-http-client/1.1   
5266b3ad-d1a7-ce42-1d06-d5387fb3513d  VSPHERE.LOCAL\Administrator                                           2023-02-27 22:52  old     10.168.194.127  Go-http-client/1.1   
52786646-6958-44ae-58f9-5a3f0ca7353c  VSPHERE.LOCAL\Administrator                                           2023-02-28 01:47  2m51s   10.168.192.18   Go-http-client/1.1   
5278b1e2-ef31-a552-031a-96d1a78aa1ce  VSPHERE.LOCAL\Administrator                                           2023-02-27 22:52  5m22s   10.168.194.127  Go-http-client/1.1   
52935591-1199-b580-4538-f18f5ea0a182  VSPHERE.LOCAL\vpxd-extension-0255c506-eb78-49d3-8115-aa73e68234ce     2023-02-02 03:11  2m27s   10.168.201.152  Go-http-client/1.1   
529837a7-eae0-22ca-4efb-8031dd6656e7  VSPHERE.LOCAL\Administrator                                           2023-02-27 22:52  old     10.168.194.127  Go-http-client/1.1   
52aa2685-3f86-cdbd-9800-874cccb548c3  VSPHERE.LOCAL\Administrator                                           2023-02-27 22:52  old     10.168.194.127  Go-http-client/1.1   
52b9ffa0-a028-8e65-97cc-e8f8cc8ed856  VSPHERE.LOCAL\Administrator                                           2023-02-27 22:52  5m21s   10.168.194.127  Go-http-client/1.1   
52da3510-15bd-52ea-6e16-4b59fdbc61ad  VSPHERE.LOCAL\vpxd-extension-0255c506-eb78-49d3-8115-aa73e68234ce     2023-02-02 03:11  8m48s   10.168.201.152  Go-http-client/1.1   
52ecac4a-2f18-c461-bae0-6be3c35ef6f4  VSPHERE.LOCAL\Administrator                                           2023-02-28 01:47  2m51s   10.168.198.98   Go-http-client/1.1
...
```
It leads to debug difficulty if one of the components has bug that can cause leaked/broken/stale sessions, we can not quickly tell which component has bug.

Setting the user agent when creating vc session should help troubleshooting vc problems if it occurs. And this should set on other components that use govomomi to talk to vc as well, including
- CAPV
- CPI vsphere
- CSI(already does)
- template-resolver-controller(from tanzu-framework)
- others that maybe missed or added in the future

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-vsphere version: all version
- Kubernetes version: (use `kubectl version`): N/A
- OS (e.g. from `/etc/os-release`):  N/A

### PR DESCRIPTION
**Pull request by adam-jian-zhang**
Set user agent for vimclient of capv "k8s-capv-useragent" to help troubleshooting.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1805

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/session/session.go [hunk 1]
```diff
@@ -188,6 +188,8 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 		return nil, err
 	}
```
Comment (srm09) [inline pkg/session/session.go:190]:
```suggestion
```

**Review by srm09 — COMMENTED** [review]
Would you be able to take screen shot of the sessions created with the user agent set to the new value? Otherwise lgtm

File: pkg/session/session.go (hunk 1 above)
Comment (adam-jian-zhang) [reply pkg/session/session.go]: done

**Comment by adam-jian-zhang** [comment]
@srm09, here is the screenshot:

capv-useragent

**Pull request merged**